Allow for specifying protocol and API base path

bellpeterm opened the pull request:
I work in an environment where the ElasticSearch API is using HTTPS and its base path is /elasticsearch so both of these improvements are needed.  I thought it might be useful for others too.

Commit:
allow for specifying protocol and API base path

s12v commented on src/main/java/me/snov/newrelic/elasticsearch/parsers/NodesStatsParser.java:
```diff
@@ -13,7 +13,7 @@ public NodesStatsParser() {
-    public NodesStatsParser(String host, int port, String username, String password) throws MalformedURLException {
-        super(NodesStats.class, new URL(HTTP, host, port, URL_CLUSTER_STATS), username, password);
+    public NodesStatsParser(String protocol, String host, int port, String basePath, String username, String password) throws MalformedURLException {
+        super(NodesStats.class, new URL(protocol, host, port, basePath + URL_CLUSTER_STATS), username, password);
```
please cover a case, when the path ends with `/`?

s12v commented on src/main/java/me/snov/newrelic/elasticsearch/ElasticsearchAgentFactory.java:
```diff
@@ -26,13 +28,21 @@ public Agent createConfiguredAgent(Map<String, Object> properties) throws Config
             throw new ConfigurationException("'host' and 'port' must be specified. Do you have a 'config/plugin.json' file?");
         }
 
+        if (protocol == null) {
```
could you please add the new options to the example configuration?

s12v reviewed (changes requested):
Thanks for the PR. Looks good, 2 minor comments and 👍

Commit:
added case for path ending in /, updated documentation

bellpeterm commented:
Great feedback.  Updated PR.

Added case to drop '/' if it is last character in basePath.
Added documentation for basePath option in line with other optional parameter.

s12v commented:
@bellpeterm, thank you. Could you please also add "protocol" to the example config?

Commit:
update example config to document protocol option.

bellpeterm commented:
@s12v my mistake.  Updated the config again in the pull request.  Cheers!

s12v commented:
thank you @bellpeterm
